presto-main/src/main/java/com/facebook/presto/spiller/FileSingleStreamSpiller.java

```diff
@@ -179,6 +179,9 @@ private void writePages(Iterator<Page> pageIterator)
     @Override
     public void close()
     {
+        if (!spillInProgress.isDone()) {
+            spillInProgress.cancel(true);
```

Comment:
@rschlussel It will get an interrupt flag set, but the task is not checking it.

Reply:
ah that's too bad.  In that case, I think it could be misleading to cancel the task since it would be a no-op. Would be better to do this in conjunction with adding proper cancellation support.

Reply:
@rschlussel Removed cancellation code to handle that in separate PR.